addons/web/static/src/env.js

```diff
@@ -78,10 +82,6 @@ export async function startServices(env) {
             await _startServices(env, toStart);
         }
     });
-    // Wait for all synchronous code so that if new services that depend on
-    // one another are added to the registry, they're all present before we
-    // start them regardless of the order they're added to the registry.
-    await Promise.resolve();
```

Comment:
This bug should probably be fixed in stable. Noticed this earlier while working on the PoS then promptly forgot about it, it can lead to services getting "double-loaded" (actually one real load and one failure because it's already started).

Reply:
what bug? here, i changed the strategy to begin the starting module process one microtask tick later, so at the end of the startModules code, I can just check if there are errors/pending modules and report them if necessary. The previous code would start modules immediately, if possible. But then, it cannot check for errors immediately

Reply:
The bug is that if we call startService and then define a service synchronously (eg because of module loading order), and the services manage to start synchronously, the service will already be present in the first call to _startServices, but we will still get the UPDATE event and try to start it again.